From: Sage Weil <sage@redhat.com>
Date: Mon, 28 Nov 2016 14:35:24 -0500
Subject: crush/CrushWrapper: drop unused 'lean' encode() argument

No callers, no users.

Signed-off-by: Sage Weil <sage@redhat.com>
(cherry picked from commit 638a38bbb55c07ad0358a35a56418e66874d1c26)

Conflicts:
	src/crush/CrushWrapper.h

[trivial conflict due to removal of write_file and read_file]

(cherry picked from commit c5f5b94fc2e18b38d05f00fec04d2d7e3c35c54c)

Resolves: rhbz#1402185
---
 src/crush/CrushWrapper.cc | 2 +-
 src/crush/CrushWrapper.h  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/crush/CrushWrapper.cc b/src/crush/CrushWrapper.cc
index 3450b48..f254ac0 100644
--- a/src/crush/CrushWrapper.cc
+++ b/src/crush/CrushWrapper.cc
@@ -1118,7 +1118,7 @@ int CrushWrapper::remove_rule(int ruleno)
   return 0;
 }
 
-void CrushWrapper::encode(bufferlist& bl, bool lean) const
+void CrushWrapper::encode(bufferlist& bl) const
 {
   assert(crush);
 
diff --git a/src/crush/CrushWrapper.h b/src/crush/CrushWrapper.h
index 1f9c878..5a8f3e9 100644
--- a/src/crush/CrushWrapper.h
+++ b/src/crush/CrushWrapper.h
@@ -1110,7 +1110,7 @@ public:
     return bl.write_file(fn);
   }
 
-  void encode(bufferlist &bl, bool lean=false) const;
+  void encode(bufferlist &bl) const;
   void decode(bufferlist::iterator &blp);
   void decode_crush_bucket(crush_bucket** bptr, bufferlist::iterator &blp);
   void dump(Formatter *f) const;
